File: src/test/java/no/capraconsulting/siren/internal/util/DatetimeTest.java
package no.capraconsulting.siren.internal.util;

import org.junit.Test;

import java.time.Instant;
import java.time.ZoneOffset;
import java.time.ZonedDateTime;

import static org.junit.Assert.assertEquals;

public class DatetimeTest {
    private static final String UTC = "2015-12-14T09:01:10.587Z";
    private static final String CET = "2015-12-14T10:01:10.587+01:00[Europe/Oslo]";

    @Test
    public void testFromInstantUTC() {
        assertEquals(UTC, Datetime.from(Instant.parse(UTC)).toString());
    }

    @Test
    public void testFromInstantCET() {
        Instant instant = ZonedDateTime.parse(CET).toInstant();
        assertEquals(ZonedDateTime.ofInstant(instant, ZoneOffset.UTC), Datetime.from(instant));
    }

    @Test
    public void testFromStringUTC() {
        assertEquals(UTC, Datetime.from(UTC).toString());
    }

    @Test
    public void testFromStringCET() {
        ZonedDateTime zonedDateTime = ZonedDateTime.parse(CET);
        assertEquals(zonedDateTime, Datetime.from(CET));
    }
}

File: src/test/kotlin/no/capraconsulting/siren/internal/util/DatetimeTest.kt
package no.capraconsulting.siren.internal.util

import java.time.Instant
import java.time.ZoneOffset
import java.time.ZonedDateTime
import org.assertj.core.api.Assertions.assertThat
import org.junit.jupiter.api.Test

private const val UTC = "2015-12-14T09:01:10.587Z"
private const val CET = "2015-12-14T10:01:10.587+01:00[Europe/Oslo]"

class DatetimeTest {

  @Test
  fun testFromInstantUTC() {
    assertThat(Instant.parse(UTC).toZonedDateTime().toString()).isEqualTo(UTC)
  }

  @Test
  fun testFromInstantCET() {
    val instant = ZonedDateTime.parse(CET).toInstant()
    assertThat(instant.toZonedDateTime())
        .isEqualTo(ZonedDateTime.ofInstant(instant, ZoneOffset.UTC))
  }

  @Test
  fun testFromStringUTC() {
    assertThat(UTC.toZonedDateTime().toString()).isEqualTo(UTC)
  }

  @Test
  fun testFromStringCET() {
    val zonedDateTime = ZonedDateTime.parse(CET)
    assertThat(CET.toZonedDateTime()).isEqualTo(zonedDateTime)
  }
}
